src/Lobby.tsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import "./App.css";
+import GameHUD from "./GameHUD";
+import { checkLogin } from "./Login";
+import { useParams } from "react-router-dom";
+import io from "socket.io-client";
+
+interface ILobby {}
+
+const Lobby = (props: ILobby) => {
+  const [loggedIn, setLoggedIn] = React.useState<boolean>(false);
+  const [users, setUsers] = React.useState<string[]>([]);
+  const [email, setEmail] = React.useState<string>("");
+  const [socket, setSocket] = React.useState(() => io("/"));
+  const [update, setUpdate] = React.useState<string>("");
+  const { id } = useParams();
+
+  React.useEffect(() => {
+    const asyncOps = async () => {
+      const loginStatus = await checkLogin();
+      if (loginStatus.logged_in) {
+        setLoggedIn(loginStatus.logged_in);
+        socket.emit("join", id, loginStatus.email);
+        setUsers((userArray) => userArray.concat([loginStatus.email]));
+        setEmail(loginStatus.email);
```

Comment:
You probably don't need `email` to be a local state variable managed with React Hooks because that email is an attribute on `currentUser` / `loginStatus`, which is already managed by state.